searchlib/src/vespa/searchlib/docstore/visitcache.cpp

```diff
@@ -92,8 +92,10 @@ CompressedBlobSet::CompressedBlobSet(const CompressionConfig &compression, const
         DataBuffer compressed;
         ConstBufferRef org = uncompressed.getBuffer();
         _compression = vespalib::compression::compress(compression, org, compressed, false);
-        _buffer.resize(compressed.getDataLen());
-        memcpy(_buffer, compressed.getData(), compressed.getDataLen());
+        _buffer = std::make_shared<vespalib::MallocPtr>(compressed.getDataLen());
+        memcpy(*_buffer, compressed.getData(), compressed.getDataLen());
```

Comment:
The `operator void*` magic in `MallocPtr` is making me a bit uneasy. Consider `_buffer->get()` to be more explicit